pkg/raft/opts.go

```diff
@@ -144,7 +170,21 @@ func NodeUrl(addr string) Option {
 	return func(opts *options) error {
 		addr = strings.TrimSpace(addr)
 		if addr == "" {
-			return errors.New("Nexus listen address must be given")
+			//try to auto determine nodeUrl from clusterUrl
+			localIps, err := getLocalIPAddress()
+			if err == nil {
+				for _, clusterUrl := range opts.clusterUrls {
+					//check if localIps contains this
+					host, _, _ := net.SplitHostPort(clusterUrl.Host)
```

Comment:
Should we check for port as well ? Since Nexus cluster can be launched locally too with different port numbers.

Reply:
We can't auto determine incase of  multiple nexus nodes are running on the same host, because the `getLocalIPAddress` would only return IP address, and we can't choose between which port the current process belongs to. In such cases one needs to explicitly pass the `nexusNodeUrl` argument.